taskrunner/Services/LotteryCleanupService.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using SCJ.Booking.Data;
+using SCJ.Booking.TaskRunner.Utils;
+using Serilog;
+using Task = System.Threading.Tasks.Task;
+
+namespace SCJ.Booking.TaskRunner.Services
+{
+    public class LotteryCleanupService
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
+
+        public LotteryCleanupService(IConfiguration configuration, ApplicationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+            _logger = LogHelper.GetLogger(configuration);
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        ///    Removes names and phone numbers from processed lottery entries
+        /// </summary>
+        public async Task RemovePersonalInfo()
+        {
+            string anonymizedName = "_anonymized";
+            string anonymizedEmail = "_anonymized@example.com";
+
+            _logger.Information("Removing personal info from lottery entries");
+
+            // calculate the oldest lottery entry to keep
+            int maxDays = _configuration.GetValue<int>(
+                "AppSettings:PurgeLotteryContactInfoAfterDays"
+            );
+            DateTime oldestDate = DateTime.UtcNow.AddDays(-maxDays);
```

Comment:
I'm not sure if it makes a difference, but I used DateTime.Now for scheduling when the lottery should run, not DateTime.UtcNow.  My code is working now properly now after I fixed SCJ-211 so it might be good to just use DateTime.Now everywhere for consistency.   The taskrunner is now running in Vancouver time using a TZ environment variable on the Openshift pod.  But in the grand scheme of things, it won't make a huge difference if this runs 7 hours too early or 7 hours too late.

Reply:
That sounds great, if it's all set up to use the server timezone 👍 I'll update that here